plugin-rapl/src/lib.rs

```diff
@@ -72,26 +99,27 @@ impl AlumetPlugin for RaplPlugin {
         }
 
         // Discover RAPL domains available in perf_events and powercap. Beware, this can fail!
-        let try_perf_events = perf_event::all_power_events();
-        let try_power_zones = powercap::all_power_zones();
+        let try_perf_events = self.get_all_power_events();
+        let try_power_zones = self.get_all_power_zones();
 
         let (available_domains, subset_indicator) = match (try_perf_events, try_power_zones) {
```

Comment:
Can you move this into the `consistency` module? No need to refactor it extensively, just extract it outside of start, in a function that returns `available_domains` and `subset_indicator`.